FEATURE_PREVIEW_DOCUMENTATION.md

```diff
@@ -0,0 +1,285 @@
+# Feature Preview Documentation
+
+This document partners with the [feature gates documentation][fg] and includes early draft documentation for features that are currently considered [alpha quality][fg-alpha].
+
+As features mature to `BETA` or `GA` state this documentation will be grown and eventually will graduate to the [main Kong Documentation][kong-docs].
```

Comment:
I don't understand the reason behind not putting feature previews on the docs site (with an appropriate maturity warning banner). I'm worried that spreading our docs across locations hurts visibility.

These notes make pretty good product documentation.

Suggestion: put these on docs.konghq.com

Reply:
I feel **very strongly** that feature gated features at an Alpha level of maturity should be kept out of the main documentation for multiple reasons:

- we don't want to make it look like we're making a commitment to the feature yet, we want to maintain full flexibility
- we don't have confidence that people read or care about maturity warnings in the docs

The `Gateway` feature is not guaranteed to land as GA in KIC and until we're _very deliberately_ doing so I believe we should keep it out of the main documentation, just as we currently keep feature gates out of the main documentation.

Reply:
Discussed offline. We'll keep documentation here until we make a decision that it's the time for a public preview of this feature.